clients/client-py/release.sh

```diff
@@ -17,17 +17,6 @@ cd "$(dirname "${0}")"
 # exit in case of bad exit code
 set -e
 
-# Local changes will not be in the release, so they should be dealt with before
-# continuing. git stash can help here! Untracked files can make it into release
-# so let's make sure we have none of them either.
-modified="$(git status --porcelain)"
-if [ -n "$modified" ]; then
-  echo "There are changes in the local tree.  This probably means"
-  echo "you'll do something unintentional.  For safety's sake, please"
-  echo 'revert or stash them!'
-  exit 66
-fi
-
```

Comment:
I'm confused about why this would be getting triggered. The guard here actually seems like something we want?

Reply:
I'm a bit confused, too.  It might be that there's some temp file generated by a process occurring at the same time.  At any rate, this is now operating in the controlled setting of a release process (and eventually in CI), so I don't think the guard is actually that useful.

Reply:
I could find out if this was easier to test, but unfortunately I don't know how to test except by making more releases..

Reply:
Let's get it to log what `git status` says and then land this.